### DISCUSSION TITLE
warning when using SetInstrumentationKey

### DISCUSSION THREAD

**Discussion by dorellanah** [IvanJosipovic/BlazorApplicationInsights#82] [Q&A] [answered]
I removed the instrumentation key from  the script and I'm setting it using SetInstrumentationKey, I've seen that the browser show this warning 
`Init.js:15 Failed to initialize AppInsights JS SDK for instance appInsights - Please provide instrumentation key`

Everything works fine, I'm just wondering if there is a way to avoid showing that warning in the console

**Comment by IvanJosipovic** [comment, marked as answer]
I have documented how to set the ConnectionString Programmatically for a Blazor WASM ASP.NET Core Hosted project. This solution will need to be manually implemented for self hosted Blazor, it essentially uses a cookie to store the ConnectionString.

I do not recommend using SetInstrumentationKey and SetConnectionString (due to https://github.com/IvanJosipovic/BlazorApplicationInsights/issues/90) and they will likely be marked as obsolete in a future version.
